src/VersionTool/Program.cs

```diff
@@ -0,0 +1,307 @@
+            {
+                var pathOption = c.Option("-p|--path", "path to search for projects", CommandOptionType.MultipleValue);
+
+                c.HelpOption("-h|--help");
+
+                c.OnExecute(() => OnList(pathOption));
+            });
+
+            var updateVersionCommand = app.Command("update-version", (c) =>
+            {
+                var pathOption = c.Option("-p|--path", "path to search for projects", CommandOptionType.MultipleValue);
+                var matchingOption = c.Option("-m|--matching", "current version to match", CommandOptionType.MultipleValue);
+
+                var versionArgument = c.Argument("version", "the version to set");
+
+                c.HelpOption("-h|--help");
+
+                c.OnExecute(() => OnUpdateVersion(pathOption, matchingOption, versionArgument));
+            });
+
+            var listDepdendency = app.Command("list-dependency", (c) =>
+            {
+                var pathOption = c.Option("-p|--path", "path to search for projects", CommandOptionType.MultipleValue);
+                var matchingOption = c.Option("-m|--matching", "current version to match", CommandOptionType.MultipleValue);
+
+                var dependencyArgument = c.Argument("dependency", "the dependency package name");
+
+                c.HelpOption("-h|--help");
+
+                c.OnExecute(() => OnListDependency(pathOption, matchingOption, dependencyArgument));
+            });
+
+            var updateDependency = app.Command("update-dependency", (c) =>
+            {
+                var pathOption = c.Option("-p|--path", "path to search for projects", CommandOptionType.MultipleValue);
+                var matchingOption = c.Option("-m|--matching", "current version to match", CommandOptionType.MultipleValue);
+
+                var dependencyArgument = c.Argument("dependency", "the dependency package name");
+                var versionArgument = c.Argument("version", "the dependency version");
+
+                c.HelpOption("-h|--help");
+
+                c.OnExecute(() => OnUpdateDependency(pathOption, matchingOption, dependencyArgument, versionArgument));
+            });
+
+
+            app.HelpOption("-h|--help");
+
+            app.OnExecute(() =>
+            {
+                app.ShowHelp();
+                return 0;
+            });
+
+            app.Execute(args);
+        }
+
+        private static int OnList(CommandOption pathOption)
+        {
+            var paths = pathOption.Values;
+            if (paths.Count == 0)
+            {
+                paths.Add(Path.GetFullPath("."));
+            }
+
+            foreach (var project in EnumerateProjects(paths))
+            {
+                Console.WriteLine($"Name:      {project.Name}");
+                Console.WriteLine($"Directory: {project.ProjectDirectory}");
+                Console.WriteLine($"Version:   {project.Version}");
+                Console.WriteLine();
```

Comment:
Nit: You can use our logging infrastructure instead of doing console writeline. You get nice colors, verbosity, etc.